ambry-store/src/main/java/com.github.ambry.store/CompactionManager.java

```diff
@@ -154,6 +154,28 @@ void controlCompactionForBlobStore(BlobStore store, boolean enable) {
     }
   }
 
+  /**
+   * Remove store from compaction manager.
+   * @param store the {@link BlobStore} to remove
+   * @return {@code true} if store is removed successfully. {@code false} if not.
+   */
+  boolean removeBlobStore(BlobStore store) {
+    boolean result;
+    if (compactionExecutor == null) {
+      stores.remove(store);
+      result = true;
+    } else if (!compactionExecutor.getStoresDisabledCompaction().contains(store)) {
+      logger.error("Fail to remove store ({}) from compaction manager because compaction of it is still enabled",
```

Comment:
This seems like it should be an IllegalStateException.  Is there a valid scenario where compaction is still enabled when removeStore is called?  If we return false and compaction is turned off later, will the store ever be removed?

Reply:
I agree with your point, however, I am trying to use same way in `scheduleNextForCompaction` method (line 360). Returning false may happen when disabling compaction on the store hasn't completed or hasn't been executed. The Helix state model will retry entire workflow (disable compaction, remove store) to guarantee the store is correctly removed.